tests/attacks/poison/test_clean_label_backdoor_attack.py

```diff
@@ -22,21 +22,26 @@
 import pytest
 
 from art.attacks.poisoning import PoisoningAttackCleanLabelBackdoor, PoisoningAttackBackdoor
-from art.attacks.poisoning.perturbations import add_pattern_bd
+from art.attacks.poisoning.perturbations import add_pattern_bd, add_pattern_bd_pytorch
 from art.utils import to_categorical
 
 from tests.utils import ARTTestException
 
 logger = logging.getLogger(__name__)
 
 
-@pytest.mark.skip_framework("non_dl_frameworks", "pytorch", "mxnet")
-def test_poison(art_warning, get_default_mnist_subset, image_dl_estimator):
+@pytest.mark.skip_framework("non_dl_frameworks", "mxnet")
+def test_poison(art_warning, get_default_mnist_subset, image_dl_estimator, framework):
     try:
         (x_train, y_train), (_, _) = get_default_mnist_subset
         classifier, _ = image_dl_estimator()
         target = to_categorical([9], 10)[0]
-        backdoor = PoisoningAttackBackdoor(add_pattern_bd)
+
+        if framework in ["pytorch", "huggingface"]:
+            backdoor = PoisoningAttackBackdoor(add_pattern_bd_pytorch)
+        else:
+            backdoor = PoisoningAttackBackdoor(add_pattern_bd)
+
```

Comment:
```suggestion
```

Reply:
changed to `backdoor = PoisoningAttackBackdoor(add_pattern_bd(channels_first=True))` for the PyTorch/Hugingface case to correctly handle the image dimensions.